### ISSUE TITLE
Macro fails in contexts that define their own Result type

### ISSUE THREAD

**Issue by dholroyd** [johnnynotsolucky/strong_id#1]
I wanted to add a StrongId in a codebase that already has a declaration like,

```
pub type Result<T> = std::result::Result<T, OurError>;
```
This is a pattern I've seen in a few rust projects.

That definition makes my use of `strong_id::strong_id!()` fail - the macro output expects but does not explicitly specify `std::result::Result` rather than this type alias.

One workaround is to wrap the strongid definition in a submodule and pub-use the result:

```
mod myid {
    use strong_id::strong_uuid;
    use std::result::Result;
    strong_uuid!(pub struct RequestId(Uuid => "req"));
}
pub use myid::RequestId;
```

Consider explicitly [qualifying](https://dev-doc.rust-lang.org/beta/reference/paths.html#path-qualifiers) `::std::result::Result` within macro-generated code.

**Comment by johnnynotsolucky** [comment]
Thanks for finding this @dholroyd! 

Looks like I missed qualifiers for Result in a few places.

I'll resolve asap.

**Comment by johnnynotsolucky** [comment]
@dholroyd this should be resolved in 0.3.0
